### ISSUE TITLE
Connecting to testnet contracts

### ISSUE THREAD

**Issue by samajammin** [clrfund/monorepo#363]
Hey folks - I'm simply trying to connect my locally running app to some testnet contracts (of a previous round).

Looking for a sanity check here to ensure I'm taking the correct steps...

1. I created an Alchemy account & set up an App on Rinkeby
2. I set up the ENV variables in /vue-app/.env:
```
VUE_APP_ETHEREUM_API_URL=https://eth-rinkeby.alchemyapi.io/v2/MY_API_KEY
VUE_APP_CLRFUND_FACTORY_ADDRESS=0x320D78bB248e019eCF23cd85E3515174Fe780bc0
VUE_APP_USER_REGISTRY_TYPE=simple
VUE_APP_RECIPIENT_REGISTRY_TYPE=optimistic
```
3. I run `yarn start:web`

When loading the frontend in my browser, I'm hit with this error:
```
vue.runtime.esm.js?e832:1888 Error: cannot estimate gas; transaction may fail or may require manual gas limit (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\": \"2.0\", \"id\": 48, \"error\": {\"code\": -32000, \"message\": \"execution reverted\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"to\":\"0x320d78bb248e019ecf23cd85e3515174fe780bc0\",\"data\":\"0xa32bf597\"},\"latest\"],\"id\":48,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://eth-rinkeby.alchemyapi.io/v2/MY_API_KEY"}, method="call", transaction={"to":"0x320D78bB248e019eCF23cd85E3515174Fe780bc0","data":"0xa32bf597"}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.0.17)
```

I can see this takes place in the /api/round.ts when trying to fetch `factory.getCurrentRound()`: https://github.com/clrfund/monorepo/blob/develop/vue-app/src/api/round.ts#L42

I can confirm that I'm otherwise able to interact with the Alchemy node, e.g. before that `factory.getCurrentRound()` call, I dropped in:
```
  provider.getBlockNumber().then(function(blockNumber) {
    console.log('Current block number: ' + blockNumber)
  })
```
This logged the correct latest block number on Rinkeby.

Looking into the request logs on Alchemy, I can see recent invalid requests, all for `eth_call` with error code `-32000`: https://docs.alchemy.com/alchemy/documentation/error-reference#standard-json-rpc-errors

![Image 2021-06-20 at 11 53 19 AM](https://user-images.githubusercontent.com/8097623/122685063-1fa79580-d1be-11eb-851e-a5df2999a438.jpg)

Any sense on why I'm hitting this gas estimation error? Thanks in advance!

**Comment by samajammin** [comment]
@auryn-macmillan perhaps you could provide some insight here? Can you please confirm if this is a correct FundingRoundFactory contract address? Is there a record of other deployed addresses I could try out?

Thanks!

**Comment by xuhcc** [comment]
Most likely the VUE_APP_CLRFUND_FACTORY_ADDRESS parameter points to a wrong contract

**Comment by samajammin** [comment]
Gotcha, thanks @xuhcc. Can you or @auryn-macmillan point me to a correct FundingRoundFactory contract address (ideally one that was used in tandem with an OptimisticRecipientRegistry 😄) ?

**Comment by auryn-macmillan** [comment]
@samajammin, I believe this is the address of the most recent Rinkeby deployment of the FundingRoundFactory.
`0x65680422D10f16f2909A8805CB1CCb2eCAB8437B`. The most recent round uses the optimistic registry.

**Comment by samajammin** [comment]
Awesome - that did the trick! Thanks.

For reference, here's my /vue-app/.env:
```
VUE_APP_ETHEREUM_API_URL=https://eth-rinkeby.alchemyapi.io/v2/MY_API_KEY
VUE_APP_BLOCK_EXPLORER=https://rinkeby.etherscan.io/tx/
VUE_APP_CLRFUND_FACTORY_ADDRESS=0x65680422D10f16f2909A8805CB1CCb2eCAB8437B
VUE_APP_RECIPIENT_REGISTRY_TYPE=optimistic
VUE_APP_USER_REGISTRY_TYPE=simple
VUE_APP_IPFS_GATEWAY_URL=https://ipfs.io
VUE_APP_GUN_API_URL=http://localhost:8765/gun
```

**Comment by samajammin** [comment]
Hmmm while the app is now clearly connecting to the factory contract (e.g. it fetches projects correctly), I'm still seeing this console error re: `UNPREDICTABLE_GAS_LIMIT`:

```
index.js?d536:166 Uncaught (in promise) Error: cannot estimate gas; transaction may fail or may require manual gas limit (error={"reason":"processing response error","code":"SERVER_ERROR","body":"{\"jsonrpc\": \"2.0\", \"id\": 54, \"error\": {\"code\": -32000, \"message\": \"execution reverted\"}}","error":{"code":-32000},"requestBody":"{\"method\":\"eth_call\",\"params\":[{\"to\":\"0x234c381028bfcef4a0ff57496f2b1be18052a199\",\"data\":\"0xaf99b63f\"},\"latest\"],\"id\":54,\"jsonrpc\":\"2.0\"}","requestMethod":"POST","url":"https://eth-rinkeby.alchemyapi.io/v2/MY_API_KEY"}, method="call", transaction={"to":"0x234C381028BfcEf4a0fF57496F2b1be18052a199","data":"0xaf99b63f"}, code=UNPREDICTABLE_GAS_LIMIT, version=providers/5.0.17)
    at Logger.makeError (index.js?d536:166)
    at Logger.throwError (index.js?d536:175)
    at checkError (json-rpc-provider.js?2b89:63)
    at StaticJsonRpcProvider.eval (json-rpc-provider.js?2b89:403)
    at Generator.throw (<anonymous>)
    at rejected (json-rpc-provider.js?2b89:6)
```

Will continue to investigate but LMK if you folks have seen this before.

**Comment by auryn-macmillan** [comment]
@samajammin I noticed that your alchemy API key is not included in the above.
`https://eth-rinkeby.alchemyapi.io/v2/MY_API_KEY` 
I assume that you replaced it before posting. But if not, I'd guess this is the source of the error.

**Comment by auryn-macmillan** [comment]
If it's not that, I'd try switching your node to infura, just to double check that it's not Alchemic specific.

Which function are you calling?

**Comment by samajammin** [comment]
Yep I am using the API key, just swapped it out when pasting the error message. I just tried switching to an Infura endpoint - unfortunately I'm still getting this same error.

Ahhh I believe figured it out - the deployed registry contracts are missing a new view method we added - the `BaseRecipientRegistry`: https://rinkeby.etherscan.io/address/0x234c381028bfcef4a0ff57496f2b1be18052a199#code#F4#L1

It's missing the `getRecipientCount` method we recently added.
https://github.com/clrfund/monorepo/blob/develop/contracts/contracts/recipientRegistry/BaseRecipientRegistry.sol#L146

Closing this out - thanks for the help 😄
